src/qa-test-eks.sh

```diff
@@ -5,12 +5,10 @@ BASEPATH=./test/integration
 ACTION=$1
 # $AWS_DEFAULT_REGION
 
-extra_cleanup () {
-    REGION=$1
-    export CLUSTERNAME=$2
+function extra_cleanup {
```

Comment:
The JQ program needs the `CLUSTERNAME` variable exported (not just local variables). Also, region is remove from this function - are we sure it's inherited from where it's called?

Reply:
I could be missing something but why is the export required? We are not using local variables in the bash script, just global? The region should be set correctly each the function is called so it's not needed to set that or the clustername again at this point

Reply:
JQ being weird. It was first fixed in this commit: https://github.com/dfds/infrastructure-modules/commit/9c55728a00cc78d44d66665bfd652262469c56cc.

If it's only a local variable, and not exported, the argument is not passed (via `--arg ROLEPREFIX "eks-${CLUSTERNAME}-"`) properly to the JQ program, meaning it won't match the roles.

I haven't tested if REGION is inherited in the extra_cleanup function.

Reply:
I'm not seeing that happen. To confirm I've just ran a test by creating an eks-qa19-rifistest role and running pre-cleanup via pipeline. It picks up that the role exists and tries to delete it so that indicates to me that it's working without needing to export.

Unfortunately that test appears to have shown a different problem:

```
An error occurred (DeleteConflict) when calling the DeleteRole operation: Cannot delete entity, must remove roles from instance profile first.
```

Which may require some extra changes to the cleanup action. I can't say why me creating that role in the web console has created an instanceprofile, mind, but one also gets created for eks-qa19-node so I'm guessing it's an AWS Service EC2 trusted entity thing. One for another pull request if it happens for real, perhaps, rather than scope creep with this one

Reply:
The JQ query consistenly failed for us, until we specifically exported the variable used with `--arg`, as @wcarlsen is my witness 👀. Not sure what have changed.

Reply:
The Prime-pipeline docker image was recently updated which would have picked up any updates to `jq` if there have been any in the base image

All I know for sure is that it's consistently working now (I've ran yet another test with no InstanceProfile or Policy attached to the role and it has been deleted by pre-cleanup) and variables are set as expected.

I'd like to merge this in based on the above discussion and test if we are happy this can be marked as resolved? I'll raise a new issue for the modification to include `remove-role-from-instance-profile` and `detach-role-policy` if any are present on the roles found